src/cryptography/x509.py

```diff
@@ -1175,6 +1207,71 @@ def public_bytes(self, encoding):
 
 
 @six.add_metaclass(abc.ABCMeta)
+class CertificateRevocationList(object):
+
+    @abc.abstractmethod
+    def fingerprint(self, algorithm):
+        """
+        Returns bytes using digest passed.
+        """
+
+    @abc.abstractproperty
+    def signature_hash_algorithm(self):
+        """
+        Returns a HashAlgorithm corresponding to the type of the digest signed
+        in the certificate.
+        """
+
+    @abc.abstractproperty
+    def issuer(self):
+        """
+        Returns the X509Name with the issuer of this CRL.
+        """
+
+    @abc.abstractproperty
+    def next_update(self):
+        """
+        Returns the date of next update for this CRL.
+        """
+
+    @abc.abstractproperty
+    def last_update(self):
+        """
+        Returns the date of last update for this CRL.
+        """
+
+    @abc.abstractproperty
+    def revoked_certificates(self):
+        """
+        Returns a list of RevokedCertificate objects for this CRL.
+        """
+
+    @abc.abstractproperty
+    def extensions(self):
+        """
+        Returns an Extensions object containing a list of CRL extensions.
+        """
+
+    @abc.abstractmethod
+    def verify(self, pubkey):
+        """
+        Verify this CRL against a given 'pubkey'.
+        """
```

Comment:
Unfortunately to properly verify a certificate or CRL you need to build a trust chain to a root/anchor, validate that none of them are expired or revoked, check that each intermediate/root is allowed to sign certificates, check that each certificate is properly signed, and potentially validate things like SCTs or signature algorithm. Complicating matters is the fact that due to cross signing many potential chains might exist and determining which one is the right one to use is a continuing problem during the 1024-bit root removal and SHA1 deprecation processes.

This process is marginally simpler in the case of CRLs where your chain is typically short, but cross signing path resolution issues can still occur (and you need to validate key usages to determine if a subordinate CRL delegate should be allowed or if it is the same as the certificate itself).